lib/developer_portal/app/controllers/developer_portal/admin/messages/trash_controller.rb

```diff
@@ -6,8 +6,10 @@ class DeveloperPortal::Admin::Messages::TrashController < DeveloperPortal::BaseC
   liquify prefix: 'messages/trash'
 
   def index
-    collection = current_account.hidden_messages.page(params[:page])
-    messages = Liquid::Drops::Message.wrap(collection)
+    collection = current_account.trashed_messages.page(params[:page])
+    # Transform deleted *received* messages to MessageRecipient instances
+    mixed_collection = collection.map { |msg| msg.sender == current_account ? msg : msg.recipients.where(receiver: current_account) }
```

Comment:
This is a quite ugly, but I think we need to know when the deleted message is a received one, because we'll need to fetch `#hidden_at` on the MessageRecipient, and  not on the associated Message.

Reply:
I checked out on this branch and I can see when deleting message from sent one it don't go to trash. 
which is want we need to do in JIRA

Reply:
Hm... weird, when I tested locally the sent messages were deleted to trash just fine.

Reply:
let me check again